pkg/block/blocktest/multipart.go

```diff
@@ -0,0 +1,302 @@
+				require.Equal(t, int(maxParts), len(listResp.Parts))
+				require.True(t, listResp.IsTruncated)
+				require.Equal(t, strconv.Itoa(int(maxParts)), *listResp.NextPartNumberMarker)
+			}
+
+			_, err = adapter.CompleteMultiPartUpload(ctx, obj, resp.UploadID, &block.MultipartUploadCompletion{
+				Part: multiParts,
+			})
+			require.NoError(t, err)
+
+			verifyListInvalid(t, ctx, adapter, obj, resp.UploadID)
+
+			getAndCheckContents(t, ctx, adapter, full, obj)
+		})
+	}
+}
+
+// Test aborting a multipart upload
+func testAdapterAbortMultipartUpload(t *testing.T, adapter block.Adapter, storageNamespace string) {
+	ctx := context.Background()
+	parts, _ := createMultipartContents(false)
+	obj, _ := objPointers(storageNamespace)
+
+	// Start an upload
+	resp, err := adapter.CreateMultiPartUpload(ctx, obj, nil, block.CreateMultiPartUploadOpts{})
+	require.NoError(t, err)
+	uploadID := resp.UploadID
+
+	uploadParts(t, ctx, adapter, obj, uploadID, parts)
+
+	listResp, err := adapter.ListParts(ctx, obj, uploadID, block.ListPartsOpts{})
+	if err == nil {
+		require.Equal(t, multipartNumberOfParts, len(listResp.Parts))
+	}
+
+	// Test abort
+	err = adapter.AbortMultiPartUpload(ctx, obj, uploadID)
+	require.NoError(t, err)
+
+	// verify no parts to list
+	verifyListInvalid(t, ctx, adapter, obj, uploadID)
+}
+
+// Test of the Multipart Copy API
+func testAdapterCopyPart(t *testing.T, adapter block.Adapter, storageNamespace string) {
+	ctx := context.Background()
+	parts, full := createMultipartContents(false)
+	obj, objCopy := objPointers(storageNamespace)
+	uploadMultiPart(t, ctx, adapter, obj, parts)
+
+	// Begin Multipart Copy
+	resp, err := adapter.CreateMultiPartUpload(ctx, objCopy, nil, block.CreateMultiPartUploadOpts{})
+	require.NoError(t, err)
+	multiParts, err := copyPart(t, ctx, adapter, obj, objCopy, resp.UploadID)
+	if adapter.BlockstoreType() == block.BlockstoreTypeAzure {
+		require.ErrorContains(t, err, "not implemented") // azurite block store emulator did not yet implement this
+		return
+	}
+	_, err = adapter.CompleteMultiPartUpload(ctx, objCopy, resp.UploadID, &block.MultipartUploadCompletion{
+		Part: multiParts,
+	})
+	require.NoError(t, err)
+
+	getAndCheckContents(t, ctx, adapter, full, objCopy)
+}
+
+// Test of the Multipart CopyRange API
+func testAdapterCopyPartRange(t *testing.T, adapter block.Adapter, storageNamespace string) {
+	ctx := context.Background()
+	parts, full := createMultipartContents(false)
+	obj, objCopy := objPointers(storageNamespace)
+	uploadMultiPart(t, ctx, adapter, obj, parts)
+
+	// Begin Multipart Copy
+	resp, err := adapter.CreateMultiPartUpload(ctx, objCopy, nil, block.CreateMultiPartUploadOpts{})
+	require.NoError(t, err)
+	multiParts := copyPartRange(t, ctx, adapter, obj, objCopy, resp.UploadID)
+	if adapter.BlockstoreType() == block.BlockstoreTypeAzure {
+		// not implemented in Azure
+		require.Nil(t, multiParts) // azurite block store emulator did not yet implement this
```

Comment:
Why do we need to verify that it's`nil`? If this test doesn't do anything when the block adapter type is Azure, why invest the time in doing any of it? I'd move this check to the top of the test and skip it (return immediately.)

This will always be `nil` until the day that someone happens to upgrade this package (probably while fixing a security issue or upgrading packages across the board for a new Go version or something). Now, they have to deal with this random failing test, which they have no context for and didn't plan for either.  
I can see merit in both approaches and the comments are helpful in this regard, so I'll leave this to your discretion. I'm fine either way.